File: frontend/src/types/stats.ts
/**
 * TypeScript types for learning statistics, streaks, and activity data.
 */
export interface StatsOverviewResponse {
  total_courses_completed: number;
  total_lessons_completed: number;
  current_streak: number;
  longest_streak: number;
  most_active_day: string | null;
  total_active_days: number;
}

export interface StreakResponse {
  current_streak: number;
  longest_streak: number;
  last_active_date: string | null;
}

export interface ActivityDayResponse {
  date: string;
  count: number;
}

export interface ActivityResponse {
  days: ActivityDayResponse[];
  total_lessons: number;
}

File: backend/app/schemas/stats.py
"""Pydantic schemas for learning statistics, streaks, and activity heatmaps."""

from pydantic import BaseModel


class StatsOverviewResponse(BaseModel):
    total_courses_completed: int
    total_lessons_completed: int
    current_streak: int
    longest_streak: int
    most_active_day: str | None  # e.g. "Monday"
    total_active_days: int


class StreakResponse(BaseModel):
    current_streak: int
    longest_streak: int
    last_active_date: str | None


class ActivityDayResponse(BaseModel):
    date: str
    count: int


class ActivityResponse(BaseModel):
    days: list[ActivityDayResponse]
    total_lessons: int
